airflow/dag_processing/manager.py

```diff
@@ -756,7 +756,10 @@ def clear_nonexistent_import_errors(self, session=NEW_SESSION):
 
             if self._file_paths:
                 query = query.where(
-                    ParseImportError.filename.notin_([f.path for f in self._file_paths]),
+                    and_(
```

Comment:
This doesn't seem right - we want to compare file/bundle combo, not just that the file doesn't exist in any bundle.